pkg/controllers/infra/cluster_controller.go

```diff
@@ -188,34 +190,34 @@ func (r *ClusterController) reconcile(ctx context.Context, infraCluster *infrav1
 	scope := scope.NewCluster(infraCluster)
 	provider := infraprovider.NewProvider(r.Client, scope)
 	if err := provider.Reconcile(ctx, infraCluster); err != nil {
-		conditions.MarkFalse(infraCluster, infrav1.InfraProviderReadyCondition, infrav1.InfraProviderProvisioningFailedReason,
+		conditions.MarkFalse(infraCluster, infrav1.InfraProviderProvisionCondition, infrav1.InfraProviderProvisionFailedReason,
 			capiv1.ConditionSeverityError, err.Error())
 		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile AWS Cluster %s/%s", infraCluster.Namespace, infraCluster.Name)
 	}
-	conditions.MarkTrue(infraCluster, infrav1.InfraProviderReadyCondition)
+	conditions.MarkTrue(infraCluster, infrav1.InfraProviderProvisionCondition)
 	infraCluster.Status.Phase = string(infrav1.ClusterPhaseInfraProvisioned)
 
 	// check Cluster status
-	if !r.clusterInitialized(ctx, infraCluster) {
+	if !provider.IsInitialized(ctx) {
 		return ctrl.Result{RequeueAfter: r.PollInterval}, nil
 	}
 
 	if err := r.reconcileCNI(ctx, infraCluster, scope); err != nil {
-		conditions.MarkFalse(infraCluster, infrav1.CNIReadyCondition, infrav1.CNIProvisioningFailedReason,
+		conditions.MarkFalse(infraCluster, infrav1.CNIProvisionCondition, infrav1.CNIProvisionFailedReason,
 			capiv1.ConditionSeverityError, err.Error())
 		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile CNI resources")
 	}
-	conditions.MarkTrue(infraCluster, infrav1.CNIReadyCondition)
+	conditions.MarkTrue(infraCluster, infrav1.CNIProvisionCondition)
 	infraCluster.Status.Phase = string(infrav1.ClusterCNIProvisioned)
 
-	if !r.clusterReady(ctx, infraCluster) {
+	if !provider.IsReady(ctx) {
 		return ctrl.Result{RequeueAfter: r.PollInterval}, nil
 	}
 	// TODO: reconcile other resources
 
 	// set ClusterResourceSet's owner reference to infraCluster
 	if err := r.reconcileClusterResourceSet(ctx, infraCluster); err != nil {
-		conditions.MarkFalse(infraCluster, capiv1.ReadyCondition, infrav1.ClusterAPIResourceProvisioningFailedReason,
+		conditions.MarkFalse(infraCluster, capiv1.ReadyCondition, infrav1.ClusterAPIResourceProvisionFailedReason,
```

Comment:
copy capiv1.ReadyCondition to our api, so that we can know all the conditions we support